### DISCUSSION TITLE
Module for ili9488

### DISCUSSION THREAD

**Discussion by namezys** [notro/panel-mipi-dbi#12] [Ideas]
I have 3.5` TFT `BIGTREETECHTFT35 SPI V2.1` that is used for 3D printers with custom linux module.

After some investigation we found that:
* This TFT is based on ILI94888
* There is custom Linux kernel that works with this TFT.
* Custom driver is used with strange name
* DST is patched

Therefore we have valid custom module for `fbtft` so it's possible to port to new DRM.

Controller Data Sheet: [ILI9488_Preliminary_DS_V090.pdf](https://github.com/notro/panel-mipi-dbi/files/14552035/ILI9488_Preliminary_DS_V090.pdf)

Linux board:
* https://biqu.equipment/collections/control-board/products/pi4b-adapter-v1-0?variant=40353646051426
* DataSheet: [BIGTREETECH CB1 User Manual.pdf](https://github.com/notro/panel-mipi-dbi/files/14552074/BIGTREETECH.CB1.User.Manual.pdf)
* Git with details: https://github.com/bigtreetech/CB1
* Git with custom Linux and kernel: https://github.com/bigtreetech/CB1-Kernel

TFT: 
* https://biqu.equipment/collections/lcd/products/bigtreetech-tft35-spi-v2-1-touchscreen-io2can-module
* DataSheet: [BIGTREETECH TFT35 SPI V2.1 User Manual.pdf](https://github.com/notro/panel-mipi-dbi/files/14552041/BIGTREETECH.TFT35.SPI.V2.1.User.Manual.pdf)
* Git with details: https://github.com/bigtreetech/TFT35-SPI/tree/master/v2
* custom module: https://github.com/bigtreetech/CB1-Kernel/blob/kernel-5.16/kernel/drivers/staging/fbtft/fb_st7796.c
* DTS: https://github.com/bigtreetech/CB1-Kernel/blob/d383db1aa4519eb9b7c7511b8a4bb8c83fa8f39d/kernel/arch/arm64/boot/dts/allwinner/sun50i-h616-biqu.dtsi#L279
* DTS overlay: https://github.com/bigtreetech/CB1-Kernel/blob/d383db1aa4519eb9b7c7511b8a4bb8c83fa8f39d/kernel/arch/arm64/boot/dts/allwinner/overlay/sun50i-h616-tft35_spi.dts#L27

I have:
* this TFT screen
* orange pi 3 with klipper

Unfortunately, I built custom kernel about 20 years ago. I'm not sure how I can do it on Armbian.
BTW I can try to port this driver but I don't know when.

**Comment by notro** [comment]
Check if your kernel have the driver module:
`$ modinfo panel-mipi-dbi`
